src/main/java/org/ibp/api/java/impl/middleware/common/validator/GermplasmUpdateValidator.java

```diff
@@ -159,4 +177,36 @@ public void validateCreationDate(final BindingResult errors, final List<Germplas
 
 	}
 
+	public void validateProgenitorsGids(final BindingResult errors, final List<GermplasmUpdateDTO> germplasmUpdateDTOList) {
+
+		// Get the gids from progenitors.
+		final Set<Integer> progenitorGids =
+			germplasmUpdateDTOList.stream().map(dto -> dto.getProgenitors().values()).flatMap(Collection::stream)
+				.filter(value -> value != null && value != 0).collect(Collectors.toSet());
+		final List<Germplasm> germplasmByGIDs = this.germplasmMiddlewareService.getGermplasmByGIDs(new ArrayList<>(progenitorGids));
```

Comment:
Can we replace usage of `this.germplasmMiddlewareService.getGermplasmByGIDs` and `this.germplasmMiddlewareService.getGermplasmByGUIDs` with `germplasmService.searchGermplasmDTO`? So that we don't need `germplasmMiddlewareService` in this class